### PR TITLE
Add ApplyGuardrailScope for security guardrail evaluation tracing

### PR DESCRIPTION
**Pull request by nikhilNava**
## Summary

Introduces a new OpenTelemetry tracing scope (\ApplyGuardrailScope\) that captures security guardrail evaluations as spans. This enables observability into content safety, policy enforcement, and risk assessment decisions made during agent operations.

### New Contracts
- **\GuardrailDetails\** — Immutable contract capturing guardian evaluation metadata: target type, decision, provider info, policy details, and content hash for forensic correlation.
- **\GuardrailFinding\** — Represents an individual risk finding with severity, category, and confidence score.
- **\GuardrailDecisionType\** — Enum for well-known guardian decisions (Allow, Block, Modify, Flag, Error).
- **\GuardrailRiskSeverity\** — String constants for risk severity levels (Low, Medium, High, Critical).
- **\GuardrailTargetType\** — Extensible type for guardrail targets (UserInput, ModelOutput, ToolCall, etc.).

### New Tracing Scope
- **\ApplyGuardrailScope\** — Disposable OpenTelemetry scope following the same pattern as \InvokeAgentScope\, \InferenceScope\, and \ExecuteToolScope\. Supports recording findings post-creation and integrates with the existing span hierarchy.

### ETW Support
- **\ApplyGuardrailData\ / \ApplyGuardrailDataBuilder\** — DTO and builder for Event Tracing for Windows logging of guardrail events.
- Updated \IA365EtwLogger\ and \A365EtwLogger\ with guardrail-specific logging methods.

### OpenTelemetry Constants
- Added \microsoft.a365.guardrail.*\ attribute keys to \OpenTelemetryConstants\.

### Tests
- Unit tests covering scope creation, finding recording, attribute emission, and edge cases.
- Integration test validating end-to-end export of guardrail spans.

## Design Notes

- Guardian spans are children of the operation they protect (e.g., inference or tool execution spans).
- Multiple guardian spans may exist under a single operation when guardrails are chained.
- Follows the existing Core + Extensions pattern and disposable scope lifecycle.

**Manual/Integration test**
below is the generated span

```
 {
   "name": "apply_guardrail Azure Content Safety llm_input",
   "traceId": "4ceed0b2561683e35af42be964263022",
   "spanId": "dd203f5da4b9be28",
   "parentSpanId": "0000000000000000",
   "kind": "Internal",
   "startTimeUnixNano": 639148092770709545,
   "endTimeUnixNano": 639148092770758119,
   "attributes": {
     "gen_ai.operation.name": "apply_guardrail",
     "gen_ai.agent.id": "03806323-fbc0-4529-9b65-403bf2401661",
     "gen_ai.agent.name": "Guardrail Agent",
     "gen_ai.agent.description": "Agent for guardrail testing.",
     "microsoft.agent.user.id": "055398b7-037e-4b92-bb1f-2e569aa14a4a",
     "microsoft.agent.user.email": "guardrailagent@ztaittest12.onmicrosoft.com",
     "microsoft.a365.agent.blueprint.id": "9d3108da-236e-464d-953e-7f4b911af7a7",
     "microsoft.tenant.id": "a70eaef2-9c8d-49e8-b735-2931dff8fe68",
     "user.id": "guardrail-caller-123",
     "user.email": "guardrail-caller@ztaitest12.onmicrosoft.com",
     "user.name": "Guardrail Caller",
     "client.address": "192.168.1.42",
     "microsoft.security.decision.type": "deny",
     "microsoft.security.target.type": "llm_input",
     "microsoft.guardian.id": "azure-content-safety-001",
     "microsoft.guardian.name": "Azure Content Safety",
     "microsoft.guardian.provider.name": "Azure",
     "microsoft.guardian.version": "2.0.0",
     "microsoft.security.target.id": "msg-12345",
     "microsoft.security.decision.reason": "Content violates hate speech policy",
     "microsoft.security.decision.code": "HATE_SPEECH_001",
     "microsoft.security.policy.id": "policy-abc",
     "microsoft.security.policy.name": "Content Safety Policy",
     "microsoft.security.policy.version": "1.2.0",
     "microsoft.security.content.input.hash": "sha256:abc123def456",
     "microsoft.security.content.modified": false,
     "microsoft.security.external_event_id": "ext-event-789",
     "microsoft.security.content.input.value": "Check this content for safety",
     "microsoft.channel.name": "msteams",
     "microsoft.channel.link": "https://guardrail-channel.link",
     "microsoft.security.content.output.value": "sanitized-hash-output"
   },
   "events": [
     {
       "name": "microsoft.security.finding",
       "timeUnixNano": 639148092770742787,
       "attributes": {
         "microsoft.security.risk.category": "hate_speech",
         "microsoft.security.risk.severity": "high",
         "microsoft.security.policy.decision.type": "deny",
         "microsoft.security.policy.id": "policy-abc",
         "microsoft.security.risk.score": 0.95,
         "microsoft.security.risk.metadata": [
           "{\"category\":\"hate\",\"confidence\":0.95}"
         ]
       }
     }
   ]
 }
```

### CODE REVIEW

File: src/Observability/Runtime/Tracing/Contracts/GuardrailRiskSeverity.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Agents.A365.Observability.Runtime.Tracing.Contracts
+{
+    /// <summary>
+    /// Well-known severity levels for security risks detected by guardrails.
+    /// </summary>
+    public static class GuardrailRiskSeverity
+    {
```
[the remark this answers is not in the document]
Comment (nikhilNava) [reply src/Observability/Runtime/Tracing/Contracts/GuardrailRiskSeverity.cs:10]: fixed

File: src/Observability/Runtime/Tracing/Scopes/ApplyGuardrailScope.cs [hunk 2]
```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.Serialization;
```
[the remark this answers is not in the document]
Comment (nikhilNava) [reply src/Observability/Runtime/Tracing/Scopes/ApplyGuardrailScope.cs]: fixed

File: src/Observability/Runtime/Tracing/Scopes/ApplyGuardrailScope.cs [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        /// Records a security finding event on the current span.
+        /// Multiple findings may be recorded per guardrail evaluation.
+        /// </summary>
+        /// <param name="finding">The security finding to record.</param>
+        public void RecordFinding(GuardrailFinding finding)
+        {
+            if (finding == null)
+            {
+                return;
+            }
```
[the remark this answers is not in the document]
Comment (nikhilNava) [reply src/Observability/Runtime/Tracing/Scopes/ApplyGuardrailScope.cs:138]: fixed

**Review by juliomenendez — APPROVED** [review]

**Review by fpfp100 — APPROVED** [review]
